server/go/pkg/server/database_test.go

```diff
@@ -32,3 +75,29 @@ func TestNewMysql(t *testing.T) {
 	assert.Nil(t, err)
 	assert.NotNil(t, dbconnection)
 }
+
+func TestSetRdbmsReplicaReadConsistencyValue(t *testing.T) {
+	dbconnection = nil
+
+	var err error
+	db, err := newMysql("root:secret@(localhost:%s)/mysql")
+	assert.Nil(t, err)
+	assert.NotNil(t, db)
+
+	// empty
+	err = setRdbmsReplicaReadConsistencyValue("")
+	assert.Nil(t, err)
+	assert.Equal(t, "", readConsistencyValue)
+	// eventual
+	err = setRdbmsReplicaReadConsistencyValue(ReplicaReadConsistencyValueEventual)
+	assert.Nil(t, err)
+	assert.Equal(t, ReplicaReadConsistencyValueEventual, readConsistencyValue)
+	// global
+	err = setRdbmsReplicaReadConsistencyValue(ReplicaReadConsistencyValueGlobal)
+	assert.Nil(t, err)
+	assert.Equal(t, ReplicaReadConsistencyValueGlobal, readConsistencyValue)
+	// session
+	err = setRdbmsReplicaReadConsistencyValue(ReplicaReadConsistencyValueSession)
+	assert.Nil(t, err)
+	assert.Equal(t, ReplicaReadConsistencyValueSession, readConsistencyValue)
```

Comment:
Given that you are asserting 4 entirely different scenarios here, can this be split into 4 tests? I recognize they would all have the same setup, but the different cases should be allowed to fail separately.

Reply:
I've split the test and cleaned up a wee bit along the way.